### ISSUE TITLE
Unexpected results with Crypto.Random

### ISSUE THREAD

**Issue by centromere** [haskell-crypto/cryptonite#127]
I ran the following program:

```haskell
module Main where

import Crypto.Random
import Data.ByteArray

main :: IO ()
main = do
  gen <- drgNew
  let (bytes, _) = randomBytesGenerate 1024 gen 
  print (bytes :: Bytes)
```

I copied the output directly in to ghci:

```
λ> let x = "..." :: String
λ> length x
8704
```

I would have expected a length of 1024.
Moreover, the output contains huge swaths of `\NUL`, leading me to believe that inappropriate areas of memory are being accessed.

**Comment by ocheron** [comment]
I think the bug is here:
https://github.com/vincenthz/hs-memory/blob/cd83e463ddd79f7f769ce0b24e1b47afedfe6592/Data/ByteArray/Bytes.hs#L157

In the `(len -# idx) ># 63#` case, `chunkLenM1` should be `64#` instead of `idx +# 64#`.

**Comment by ocheron** [comment]
PR created: vincenthz/hs-memory#27.

**Comment by vincenthz** [comment]
It's now fixed in memory 0.14.1.
